mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -114,21 +106,47 @@ class DenseMatrix(val numRows: Int, val numCols: Int, val values: Array[Double])
   require(values.length == numRows * numCols, "The number of values supplied doesn't match the " +
     s"size of the matrix! values.length: ${values.length}, numRows * numCols: ${numRows * numCols}")
 
-  override def toArray: Array[Double] = values
+  override def toArray: Array[Double] = {
```

Comment:
Note that we assumed this is cheap for dense matrices before. Could you check its usage?